yarn-project/noir-contracts/src/contracts/noir-aztec3/src/lib.nr

```diff
@@ -1,17 +1,21 @@
 mod abi;
 mod context;
+mod messaging;
 mod notes;
 mod state_vars;
 mod types;
 mod utils;
 mod private_call_stack_item;
 mod public_call_stack_item;
 mod public_call;
+mod emit_data;
 
 global Nonce = 1;
 global NoteHash = 2;
 global NoteStorageSlot = 3;
 global MappingStorageSlot = 4;
 global Nullifier = 5;
+global L1ToL2MessageSecret = 29;
```

Comment:
I am out of the loop - why is this 29? is this the pedereson generator constant?

Reply:
Its index 29 here: https://github.com/AztecProtocol/aztec-packages/blob/35b3b7ca92625ffca50e301bd4b3aa4d07843cd9/circuits/cpp/src/aztec3/constants.hpp#L89